tests/functional/test_rose_stem.py

```diff
@@ -228,316 +164,261 @@ def setup_stem_repo(tmp_path_factory, monkeymodule, request):
         'stem_sources': [
             str(setup_stem_repo['workingcopy']), "fcm:foo.x_tr@head"
         ],
         'workflow_name': setup_stem_repo['suitename']
     }
-    yield await rose_stem_run_template(rose_stem_opts)
-
-
-@pytest.mark.parametrize(
-    'expected',
-    [
-        "run_ok",
-        "RUN_NAMES=['earl_grey', 'milk', 'sugar', 'spoon', 'cup', 'milk']",
-        "SOURCE_FOO=\"{workingcopy} fcm:foo.x_tr@head\"",
-        "HOST_SOURCE_FOO=\"{hostname}:{workingcopy} fcm:foo.x_tr@head\"",
-        "SOURCE_FOO_BASE=\"{workingcopy}\"\n",
-        "SOURCE_FOO_BASE=\"{hostname}:{workingcopy}\"\n",
-        "SOURCE_FOO_REV=\"\"\n",
-        "SOURCE_FOO_MIRROR=\"fcm:foo.xm/trunk@1\"\n",
-    ]
-)
-def test_basic(rose_stem_run_basic, expected):
-    """Check that assorted variables have been exported.
-    """
-    if expected == 'run_ok':
-        assert rose_stem_run_basic['run_stem'].returncode == 0
-    else:
-        expected = expected.format(
-            workingcopy=rose_stem_run_basic['workingcopy'],
-            hostname=HOST
-        )
-        assert expected in rose_stem_run_basic['jobout_content']
+    template_vars = await rose_stem_runner(rose_stem_opts)
+
+    workingcopy = setup_stem_repo["workingcopy"]
+    check_template_variables(
+        {
+            "RUN_NAMES":
+                "['earl_grey', 'milk', 'sugar', 'spoon', 'cup', 'milk']",
+            "SOURCE_FOO": f'"{workingcopy} fcm:foo.x_tr@head"',
+            "HOST_SOURCE_FOO": f'"{HOST}:{workingcopy} fcm:foo.x_tr@head"',
+            "SOURCE_FOO_BASE": f'"{workingcopy}"',
+            "HOST_SOURCE_FOO_BASE": f'"{HOST}:{workingcopy}"',
+            "SOURCE_FOO_REV": '""',
+            "SOURCE_FOO_MIRROR": '"fcm:foo.xm/trunk@1"',
+        },
+        template_vars,
+    )
 
 
-@pytest.fixture()
-async def project_override(
-    rose_stem_run_template, setup_stem_repo
-):
+async def test_manual_project_override(rose_stem_runner, setup_stem_repo):
+    """It should accept named project sources.
+
+    https://github.com/metomi/rose/blob/2c8956a9464bd277c8eb24d38af4803cba4c1243/t/rose-stem/00-run-basic.t#L80-L112
+    """
     rose_stem_opts = {
-        'stem_groups': ['earl_grey', 'milk,sugar', 'spoon,cup,milk'],
-        'stem_sources': [
-            f'bar={str(setup_stem_repo["workingcopy"])}', "fcm:foo.x_tr@head"
+        "stem_groups": ["earl_grey", "milk,sugar", "spoon,cup,milk"],
+        "stem_sources": [
+            # specify a named source called "bar"
+            f'bar={setup_stem_repo["workingcopy"]}',
+            "fcm:foo.x_tr@head",
         ],
-        'workflow_name': setup_stem_repo['suitename']
+        "workflow_name": setup_stem_repo["suitename"],
     }
-    yield await rose_stem_run_template(rose_stem_opts)
-
-
-@pytest.mark.parametrize(
-    'expected',
-    [
-        "run_ok",
-        (
-            "RUN_NAMES=[\'earl_grey\', \'milk\', \'sugar\', "
-            "\'spoon\', \'cup\', \'milk\']"
```

Comment:
Normalized strings ala Black which says we should switch quote style in preference to escaping (as does pep8 I think).